content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+
+   ```json
+   "MY_DISCOVERY_DOMAIN:sub": "spiffe://example.org/ns/default/sa/default"
+   ```
+   But substitute your Discovery Provider FQDN for MY\_DISCOVERY\_DOMAIN. Add a comma at the end of the previous line after `"mys3"`. When finished, the JSON should look similar to:
+
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Statement": [
+       {
+         "Effect": "Allow",
+         "Principal": {
+           "Federated": "arn:aws:iam::012345678901:oidc-provider/MY_DISCOVERY_DOMAIN"
+         },
+         "Action": "sts:AssumeRoleWithWebIdentity",
+         "Condition": {
+           "StringEquals": {
+             "MY_DISCOVERY_DOMAIN:aud": "mys3",
+             "MY_DISCOVERY_DOMAIN:sub": "spiffe://example.org/ns/default/sa/default"
+           }
+         }
+       }
+     ]
+   }
+   ```
+
+5. Click **Update Trust Policy**. This change to the IAM role takes a minute or two to propagate.
+
+## Create an AWS S3 Test File
+
+Create a simple test file in an AWS S3 bucket to use for testing.
+
+1. Create a text file on your local computer called `test.txt` with the following single line:
+
+   ```console
+   oidc-tutorial file
+   ```
+
+   You'll upload this file to the S3 bucket momentarily.
+
+2. Navigate to the AWS [Amazon S3 page](https://s3.console.aws.amazon.com/s3/home), logging in if necessary.
+
+3. Click **Create bucket**.
+
+4. Under **Bucket name** type `oidc-federation-test-bucket`.
+
+5. Leave the **Region**, **Bucket settings for Block Public Access**, and **Advanced settings** at the default values and click **Create bucket**.
+
+6. Click `oidc-federation-test-bucket` to open the bucket. 
+
+7. Click **Upload**.
+
+8. Add the `test.txt` file to the upload area using your local file navigator or drag and drop.
+
+9. Click **Upload**.
+
+# Part Three: Test Access to AWS S3
+
+Now that Kubernetes and AWS are configured for OIDC federation, we'll test the connection. This test retrieves a JWT SVID from the SPIRE Agent and uses the JWT token in the JWT SVID to access S3.
+
+1. Run the following command to list the full name of the client pod:
+   ```console
+   $ kubectl get pods
+
+   NAME                      READY   STATUS    RESTARTS   AGE
+   client-74d4467b44-7nrs2   1/1     Running   0          27s
+   ```
+
+2. Start a shell on the client using the client pod name shown under **NAME** in the output of the last step.
+
+   ```console
+   $ kubectl exec -it client-74d4467b44-7nrs2 /bin/sh
+   ```
+
+3. Fetch a JWT SVID from the identity provider on AWS and save the token from the JWT SVID into a file on the client container called `token`:
+
+   ```console
+   # /opt/spire/bin/spire-agent api fetch jwt -audience mys3 -socketPath /run/spire/sockets/agent.sock | sed '2!d' | sed 's/[[:space:]]//g' > token
+   ```
```

Comment:
There's a lot that can go wrong at this step. Consider asking the user to verify the contents of the token file to make sure it was written.

Reply:
I've added a step to cat the token file.